lib/router.js

```diff
@@ -36,10 +36,13 @@ Cherrytree.prototype.initialize = function (options) {
   this.middleware = []
   this.options = _.extend({
     interceptLinks: true,
-    logError: true
+    logError: true,
+    Promise: global.Promise
   }, options)
   this.log = createLogger(this.options.log)
   this.logError = createLogger(this.options.logError, true)
+
+  invariant(typeof this.options.Promise === 'function', 'Cherrytree requires an ES6 Promise implementation, either as an explicit option or a global Promise')
```

Comment:
invariant is not an es6 thing, but just a way of adding runtime assertions to the code in a slightly less verbose manner